media/js/firefox/new.js

```diff
@@ -43,34 +45,10 @@
         document.dispatchEvent(event);
     };
 
-    var uiTourGetConfiguration = function(configName, callback) {
-        uiTourSendEvent('getConfiguration', {
-            configuration: configName,
-            callbackID: function() {
-                var id = Math.random().toString(36).replace(/[^a-z]+/g, '');
-
-                function listener(event) {
-                    if (typeof event.detail === 'object' && event.detail.callbackID === id) {
-                        document.removeEventListener('mozUITourResponse', listener);
-                        callback(event.detail.data);
-                    }
-                }
-
-                document.addEventListener('mozUITourResponse', listener);
-
-                return id;
-            }.call()
-        });
-    };
-
-    if (window.isFirefox()) {
-        // data-latest-firefox includes point release information
-        var latestFirefoxVersionFull = $html.attr('data-latest-firefox');
-
-        // get latest full version (no point release info) for initial check
-        var latestFirefoxVersion = parseInt(latestFirefoxVersionFull.split('.')[0], 10);
-
-        if (window.isFirefoxUpToDate(latestFirefoxVersion + '')) {
+    if (client.isFirefox && !client.isFirefoxiOS) {
+        // Detect whether the Firefox is up-to-date in a non-strict way. The minor version and channel are not
+        // considered. This can/should be strict, once the UX especially for ESR users is decided. (Bug 939470)
+        if (client._isFirefoxUpToDate(false)) {
```

Comment:
I like they way this is setting us up for a simple change in the near future :+1: